### DISCUSSION TITLE
Need help with a policy

### DISCUSSION THREAD

**Discussion by monks1975** [supabase/supabase#10998] [Questions] [answered]
I have a 'classifiers' table with RLS and I want to only allow updates if there exists a row in another table 'classifiers_users' who have membership of the classifier. classifiers_users has columns- id, is_manager, classifier_id and user_id. I'm trying and failing to get this working, anyone put me in the right direction? This is my attempt:

```sql
CREATE POLICY "update_for_authenticated_members_only"
ON public.classifiers
FOR UPDATE
TO authenticated

WITH CHECK (
  EXISTS(
      SELECT
        *
      FROM
        public.classifiers_users cu
      WHERE
        (
          cu.classifier_id = id
          AND cu.user_id = uid()
        )
    )
);
```

**Comment by GaryAustin1** [comment]
Update RLS requires a USING policy.  The WITH CHECK policy is optional and will be the same as USING if not there.
You also need a SELECT policy and meet it.
https://www.postgresql.org/docs/current/sql-createpolicy.html
(Not looking at your actual logic)

**Reply by monks1975** [reply]
Thanks @GaryAustin1 I managed to get it half working with a USING expression like you say, checking cu.user_id against uid() works, but how do I get the id of the row being checked by RLS to query against cu.classifier_id? When I try to use id in my expression in the supabase policy editor, it changes it to cu.id (presumably because I am inside a SELECT from classifiers_users).

**Comment by monks1975** [comment, marked as answer]
Explicitly specifying classifiers.id solved the last issue, and I used a slightly different query as suggested by @GaryAustin1 . Here's what worked for me in the end:

``` sql
CREATE POLICY "update_for_authenticated_members_only"
ON public.classifiers
FOR UPDATE
TO authenticated

-- return a boolean if a row exists in classifiers_users showing user membership of classifier

USING (
  EXISTS (
      SELECT
        clsmembs.user_id,
        clsmembs.classifier_id
      FROM
        classifiers_users clsmembs
      WHERE
        (
          (clsmembs.classifier_id = classifiers.id)
          AND (clsmembs.user_id = uid())
        )
    )
)
```
